rust/chains/tw_polkadot/src/extrinsic.rs

```diff
@@ -0,0 +1,554 @@
+
+        let data = ExtrinsicEncoder::new()
+            .add(call_index)
+            .add(wu.slashing_spans as u32)
+            .finalize();
+
+        Ok(data)
+    }
+
+    fn encode_staking_nominate(&self, n: &Nominate) -> EncodeResult<Vec<u8>> {
+        let mut data = Vec::new();
+
+        // Encode call index
+        let call_index =
+            self.get_custom_call_index_or_network(STAKING_NOMINATE, &n.call_indices)?;
+        call_index.to_scale_into(&mut data);
+
+        // Encode account IDs for nominators
+        Compact(n.nominators.len()).to_scale_into(&mut data);
+        let raw = self.should_encode_raw_account();
+        n.nominators
+            .iter()
+            .map(|s| {
+                let addr = SS58Address::from_str(s).map_err(|_| EncodeError::InvalidAddress)?;
+                if !raw {
+                    data.push(0x00);
+                }
+                addr.to_scale_into(&mut data);
+                Ok(())
+            })
+            .try_for_each(identity)?;
+
+        Ok(data)
+    }
+
+    fn encode_staking_chill(&self, c: &Chill) -> EncodeResult<Vec<u8>> {
+        // Encode call index
+        let call_index = self.get_custom_call_index_or_network(STAKING_CHILL, &c.call_indices)?;
+
+        Ok(call_index.to_scale())
+    }
+
+    fn encode_staking_chill_and_unbond(&self, cau: &ChillAndUnbond) -> EncodeResult<Vec<u8>> {
+        let first = self.encode_staking_call(&Staking {
+            message_oneof: StakingVariant::chill(Chill {
+                call_indices: cau.call_indices.clone(),
+            }),
+        })?;
+
+        let second = self.encode_staking_call(&Staking {
+            message_oneof: StakingVariant::unbond(Unbond {
+                value: cau.value.clone(),
+                call_indices: cau.call_indices.clone(),
+            }),
+        })?;
+
+        // Encode both calls as batched
+        self.encode_batch(&[first, second], &cau.call_indices)
+    }
+
+    fn encode_staking_call(&self, s: &Staking) -> EncodeResult<Vec<u8>> {
+        match &s.message_oneof {
+            StakingVariant::bond(b) => self.encode_staking_bond(b),
+            StakingVariant::bond_and_nominate(ban) => self.encode_staking_bond_and_nominate(ban),
+            StakingVariant::bond_extra(be) => self.encode_staking_bond_extra(be),
+            StakingVariant::unbond(u) => self.encode_staking_unbond(u),
+            StakingVariant::withdraw_unbonded(wu) => self.encode_staking_withdraw_unbonded(wu),
+            StakingVariant::nominate(n) => self.encode_staking_nominate(n),
+            StakingVariant::chill(c) => self.encode_staking_chill(c),
+            StakingVariant::chill_and_unbond(cau) => self.encode_staking_chill_and_unbond(cau),
+            StakingVariant::rebond(r) => self.encode_staking_rebond(r),
+            StakingVariant::None => Ok(Vec::new()),
+        }
+    }
+
+    fn encode_polymesh_join_identity_as_key(&self, j: &JoinIdentityAsKey) -> EncodeResult<Vec<u8>> {
```

Comment:
I think it's worth to move all Polymesh specific features to a separate `chains/tw_polymesh` crate-blockchain implementation. However, we can do that at the next iteration.